src/components/AppSidebar/AppSidebar.vue

```diff
@@ -1,31 +1,308 @@
 <!--
- - @copyright Copyright (c) 2019 Christoph Wurst <christoph@winzerhof-wurst.at>
- -
- - @author Christoph Wurst <christoph@winzerhof-wurst.at>
- -
- - @license GNU AGPL version 3 or any later version
- -
- - This program is free software: you can redistribute it and/or modify
- - it under the terms of the GNU Affero General Public License as
- - published by the Free Software Foundation, either version 3 of the
- - License, or (at your option) any later version.
- -
- - This program is distributed in the hope that it will be useful,
- - but WITHOUT ANY WARRANTY; without even the implied warranty of
- - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- - GNU Affero General Public License for more details.
- -
- - You should have received a copy of the GNU Affero General Public License
- - along with this program. If not, see <http://www.gnu.org/licenses/>.
- -
- -->
+  - @copyright Copyright (c) 2019 John Molakvoæ <skjnldsv@protonmail.com>
+  -
+  - @author John Molakvoæ <skjnldsv@protonmail.com>
+  -
+  - @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -
+  -->
 
 <template>
-	<div id="app-sidebar">
-		<slot />
-	</div>
+	<transition name="slide-right">
+		<aside id="app-sidebar">
+			<header class="app-sidebar-header">
+				<a href="#" class="icon-close" :title="t('core', 'close')"
+					@click="closeSidebar" />
+				<div class="app-sidebar-header__figure" :style="{
+					backgroundImage: `url(${background})`
+				}">
+					<slot class="app-sidebar-header__background" name="header-background" />
```

Comment:
Is this really about the background of the header? Seems to be more like the foreground of the header (?)

Reply:
This is about the illustration on the top of the sidebar (file preview, nextcloud talk conversation... etc) :)
How do you think we should call it? :thinking:

Reply:
Maybe "teaser"/"header-teaser" or just "header"?

Reply:
Good for me :)